rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbCreateAlarmNode.java

```diff
@@ -190,6 +191,24 @@ private Alarm getAlarmFromMessage(TbContext ctx, TbMsg msg) throws IOException {
         return Futures.transform(asyncUpdated, a -> new TbAlarmResult(false, true, false, a), MoreExecutors.directExecutor());
     }
 
+    private void updateStatus(Alarm existingAlarm, Alarm msgAlarm) {
+        boolean isActive = existingAlarm.getStatus() == AlarmStatus.ACTIVE_ACK
+                || existingAlarm.getStatus() == AlarmStatus.ACTIVE_UNACK;
+
+        boolean acknowledged = existingAlarm.getStatus() == AlarmStatus.ACTIVE_ACK
+                || existingAlarm.getStatus() == AlarmStatus.CLEARED_ACK;
+        boolean shouldBeAcknowledged = msgAlarm.getStatus() == AlarmStatus.ACTIVE_ACK
+                || msgAlarm.getStatus() == AlarmStatus.CLEARED_ACK;
+
+        if (!acknowledged && shouldBeAcknowledged) {
+            if (isActive) {
+                existingAlarm.setStatus(AlarmStatus.ACTIVE_ACK);
+            } else {
+                existingAlarm.setStatus(AlarmStatus.CLEARED_ACK);
+            }
+        }
+    }
```

Comment:
Maybe refactor to smth like
```
if (!existingAlarm.getStatus().isAck() && msgAlarm.getStatus().isAck()) {
  if (!existingAlarm.getStatus().isCleared()) {
    newStatus = ACTIVE_ACK;
  } else {
    newStatus = CLEARED_ACK;
}
```
And also, what if the status of existing alarm is `ACTIVE_UNACK` and status in the msg is `CLEARED_ACK`? Maybe we'd need to give an error if existing and new status have different clear status, as there is another node for alarm clearing.